src/app/components/details/details.rs

```diff
@@ -254,16 +246,29 @@ impl Details {
 
             if let Some(art) = album.art.clone() {
                 let widget = self.widget.downgrade();
+                let header = self.widget.header_widget().downgrade();
+                let header_mobile = self.widget.header_mobile().downgrade();
                 let modal = self.modal.downgrade();
 
                 self.worker.send_local_task(async move {
                     let pixbuf = ImageLoader::new()
                         .load_remote(&art[..], "jpg", 200, 200)
                         .await;
-                    if let (Some(widget), Some(modal), Some(ref pixbuf)) =
-                        (widget.upgrade(), modal.upgrade(), pixbuf)
-                    {
-                        widget.set_artwork(pixbuf);
+                    if let (
+                        Some(widget),
+                        Some(modal),
+                        Some(header),
+                        Some(header_mobile),
+                        Some(ref pixbuf),
+                    ) = (
+                        widget.upgrade(),
+                        modal.upgrade(),
+                        header.upgrade(),
+                        header_mobile.upgrade(),
+                        pixbuf,
+                    ) {
+                        header.set_artwork(pixbuf);
```

Comment:
might be worth adding a set_artwork fn to the details widget, so that you dont  have to upgrade two headers here as well